presidio_evaluator/evaluator_2/evaluator.py

```diff
@@ -0,0 +1,224 @@
+from collections import Counter
+from copy import deepcopy
+from typing import List, Tuple, Dict
+
+from tqdm import tqdm
+
+from presidio_evaluator import Span
+from presidio_evaluator.evaluator_2 import (TokenOutput,
+                                            SpanOutput,
+                                            ModelPrediction,
+                                            SampleError,
+                                            EvaluationResult)
+
+
+class Evaluator:
+    def __init__(
+            self,
+            entities_to_keep: List[str],
+            compare_by_io: bool = True,
+            entity_mapping: Dict[str, str] = None
+    ):
+        """
+        Evaluate a PII detection model or a PII model.
+        :param entities_to_keep: List of entity names to focus the evaluator on
+        (and ignore the rest).
+        Default is None = all entities. If the provided model has a list of
+        entities to keep, this list would be used for evaluation.
+        :param compare_by_io: True if comparison should be done on the entity
+        level and not the sub-entity level
+        :param entity_mapping: A dictionary of entity names to map between
+        input dataset and the entities from PII model
+        """
+        self.compare_by_io = compare_by_io
+        self.entities_to_keep = entities_to_keep
+        self.entity_mapping = entity_mapping
+
+    def compare_token(self, annotated_tokens: List[str],
+                      predicted_tokens: List[str]) -> Tuple[List[TokenOutput], Counter]:
+        """
+        Compares ground truth tags (annotation) and predicted (prediction)
+        at token level.
+        :param annotated_tokens: truth annotation tokens from InputSample
+        :param predicted_tokens: predicted tokens from PII model/system
+        :returns: a list of TokenOutput and a list of objects of type Counter
+        with structure {(actual, predicted) : count}
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def compare_span(
+            annotated_spans: List[Span], predicted_spans: List[Span]
```

Comment:
We are using the same approach as sem eval 13, right? If yes, consider putting a link which describes the evaluation strategy